run/secret_manager/main.tf

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "google_service_account" "default" {
+  account_id   = "cloud-run-service-account"
+  display_name = "Service account for Cloud Run"
+}
+
+resource "google_secret_manager_secret" "default" {
+  secret_id = "my-secret"
+  replication {
+    automatic = true
+  }
+}
+
+resource "google_secret_manager_secret_version" "default" {
+  secret      = google_secret_manager_secret.default.name
+  secret_data = "this is secret data"
+}
+
+resource "google_secret_manager_secret_iam_member" "default" {
+  secret_id = google_secret_manager_secret.default.id
+  role      = "roles/secretmanager.secretAccessor"
+  # Grant the default Compute service account access to this secret.
+  member     = "serviceAccount:${google_service_account.default.email}"
+  depends_on = [google_secret_manager_secret.default]
+}
+
+resource "google_cloud_run_v2_service" "default" {
+  name     = "cloudrun-srv"
+  location = "us-central1"
+  ingress  = "INGRESS_TRAFFIC_ALL"
+
+  template {
+    volumes {
```

Comment:
Since you can mount secrets as volumes, like you have, but you can also set them as environment variables, should we include both styles in the same service definition? Or have two services in the same main.tf using the same secret in different ways, with separate region tags for ease of inclusion in the docs?

Reply:
Since they would both be a sample achieving the same goal in 2 different ways, I think that yes we can include them in the same sample file.  Will add.